Mage/src/main/java/mage/players/Player.java

```diff
@@ -105,7 +106,11 @@ default boolean isComputer() {
 
     void addAbility(Ability ability);
 
-    Counters getCounters();
+    /**
+     * The counters should be manipulated (adding or removing counters) with the appropriate addCounters/removeCounters/getCounterCount methods.
+     * This returns a copy for specific usage, to make sure the Player's counters are not altered from there.
+     */
+    Counters getCopyCounters();
```

Comment:
I don't love this method name `getCopyCounters()`, maybe `getCountersAsCopy()` or `getCountersCopied()` ?